### PR TITLE
Add Zstandard (zstd) support

### PR DESCRIPTION
**Pull request by mtb-xt**
Simple edit to add [zstd](https://github.com/facebook/zstd) support, using `--adapt` flag (automatically adjusts compression level based on IO)

### CODE REVIEW

**Comment by yboetz** [comment]
Hi. I was a bit busy last week, so sorry for the late answer. Thanks for the PR though!

Can you change the PR to merge into the develop branch?

File: pyznap/ssh.py [hunk 1]
```diff
@@ -131,7 +131,8 @@ def setup_compression(self, _type):
                  'bzip2': (['bzip2'], ['bzip2', '-dfc']),
                  'pigz': (['pigz'], ['pigz', '-dc']),
                  'xz': (['xz'], ['xz', '-d']),
-                 'lz4': (['lz4'], ['lz4', '-dc'])}
+                 'lz4': (['lz4'], ['lz4', '-dc']),
+                 'zstd': (['zstd --adapt'], ['unzstd'])}
```
Comment (yboetz) [inline pyznap/ssh.py]: This should be `['zstd', '--adapt']` to be consistent.

**Comment by mtb-xt** [comment]
> Hi. I was a bit busy last week, so sorry for the late answer. Thanks for the PR though!
> 
> Can you change the PR to merge into the develop branch?

I made the necessary changes, could you please check?

P.S. Thank you for writing this simple and elegant utility!

**Comment by yboetz** [comment]
Thanks for the change. I quickly want to check how zstd compression works later, then I'll approve.

**Comment by yboetz** [comment]
Looks good, thanks! I'll make a new release later this week.

**Pull request merged**